src/core/etl/src/Flow/ETL/ExternalSort/MemorySort.php

```diff
@@ -64,6 +64,6 @@ public function sortBy(EntryReference ...$refs) : Extractor
 
         $this->cache->clear($this->cacheId);
 
-        return new Extractor\ProcessExtractor(...$mergedRows->sortBy(...$refs)->chunks($maxSize));
+        return new Extractor\ProcessExtractor(...\iterator_to_array($mergedRows->sortBy(...$refs)->chunks($maxSize)));
```

Comment:
hmm, can't we return here `$mergedRows->sortBy(...$refs)->chunks($maxSize)` which is now a generator?

Reply:
You cannot unpack a generator in PHP 😞

Reply:
But it should be pretty straightforward to create GeneratorExtractor that would accept a Generator in the constructor and just use it to yield rows in the extract method which I think should reduce the memory consumption, no?

Reply:
Added new extractor.